src/main/jbake/content/blog/2017/07/opentracing-ejb.adoc

```diff
@@ -0,0 +1,397 @@
+
+`src/main/java/io/opentracing/contrib/ejb/example/InventoryNotificationService.java`
+```java
+package io.opentracing.contrib.ejb.example;
+
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.util.GlobalTracer;
+
+import javax.ejb.Asynchronous;
+import javax.ejb.Stateless;
+import java.util.logging.Logger;
+
+/**
+ * This is the final call in the chain. This is an asynchronous stateless EJB, which obtains the span context
+ * via a method parameter. This bean is not intercepted in any way by us, so, the span context received is exactly
+ * the same as what was sent by the caller.
+ *
+ * @author Juraci Paixão Kröhling
+ */
+@Stateless
+@Asynchronous
+public class InventoryNotificationService {
+    private static final Logger log = Logger.getLogger(InventoryNotificationService.class.getName());
+
+    public void sendNotification(SpanContext context) {
+        Span span = GlobalTracer.get().buildSpan("sendNotification").asChildOf(context).startManual();
+        log.info("Sending an inventory change notification");
+        span.finish();
+    }
+
+}
+```
+
+Now, let's do a final sanity check and see if everything is in the right place:
+`mvn wildfly-swarm:run` . As before, the final message should be `WildFly Swarm is Ready`. Hit `Ctrl+C` and let's
+setup our tracing backend.
+
+== The tracing backend
+
+Instrumenting our code is one part of the story. The other part is to plug in an actual OpenTracing implementation that
+is capable of capturing the spans and submitting them to a backend service. For our demo, we'll use Jaeger. If you don't
+have a Jaeger server running yet, one can be started via Docker as follows:
+
+```bash
+docker run \
+    --rm \
+    -p5775:5775/udp \
+    -p6831:6831/udp \
+    -p6832:6832/udp \
+    -p5778:5778 \
+    -p16686:16686 \
+    -p14268:14268 \
+    --name jaeger \
+    jaegertracing/all-in-one:latest
+```
+
+== Tying everything together
+
+Now that we have our code ready and a tracing backend, let's start Wildfly Swarm passing a service name, which is the only
+property required by the Jaeger client. By default, Jaeger's Java tracer will attempt to send traces via UDP to a 
+Jaeger Agent located on the local machine. If you are using a different architecture, refer to the 
+link:https://github.com/uber/jaeger-client-java/tree/master/jaeger-core[Jaeger's documentation]
+on how to use environment variables to configure the client, or refer to the 
+link:https://github.com/wildfly-swarm/wildfly-swarm-examples/tree/master/jaeger/jaeger-servlet[Jaeger's fraction for Wildfly Swarm].
+
+For our demo, we'll use a property that is recognized by the Jaeger's Wildfly Swarm Fraction. The other two properties are telling
+Jaeger that every request should be sampled.
+
+```bash
+mvn wildfly-swarm:run -Dswarm.jaeger.service-name=order-processing -Dswarm.jaeger.sampler-type=const -Dswarm.jaeger.sampler-parameter=1
```

Comment:
@jpkrohling Never mind, looks like this is my fault.  Somehow when I generated the project I didn't get the proper dependencies.